### DISCUSSION TITLE
optimize single objective with many constraints

### DISCUSSION THREAD

**Discussion by kevinxiaomu** [optuna/optuna#4224] [Q&A] [answered]
I would like to optimize about 60 design prameters to get a software generated errorFunction minimized, but the system is subject to many strict constaints (more than 30, each contraint is some software output related with the design parameters), is optuna a suitable tool that can be used?

**Comment by sami-ka** [comment, marked as answer]
Hi, I discovered recently that you can technically integrate constraints for the parameters sampling.

> https://optuna.readthedocs.io/en/stable/faq.html#how-can-i-optimize-a-model-with-some-constraints

For my problem, I ended up changing my formulation in order to not integrate constraints so I haven't tried it yet though.
I don't know with a lot of parameters and constraints if it would be efficient. Maybe working on the model formulation will decrease the number of constraints you have?

**Comment by gen740** [comment]
TBH, optuna is not good at optimizing far many parameters (and constraints).

[The FAQ](https://optuna.readthedocs.io/en/stable/faq.html#how-can-i-optimize-a-model-with-some-constraints) is a bit outdated, so you can use the `constraints feature` in `TPESampler` either. It might be a good idea to use a `TPESampler` and set a `trials` large number.

Ideally, if you can reduce the number of parameters, do it.
